spec/system/admin_customize_themes_config_area_spec.rb

```diff
@@ -37,11 +42,11 @@
 
   it "allows to mark theme as active" do
     config_area.visit
-    expect(config_area).to have_badge(theme, "--active")
+    expect(config_area).to have_badge(default_theme, "--active")
     expect(config_area).to have_no_badge(theme_2, "--active")
     config_area.mark_as_active(theme_2)
     expect(config_area).to have_badge(theme_2, "--active")
-    expect(config_area).to have_no_badge(theme, "--active")
+    expect(config_area).to have_no_badge(foundation_theme, "--active")
```

Comment:
If the foundation theme is the default theme, it's not possible to remove it from being the default theme since `Theme.clear_default!` resets `SiteSetting.default_theme_id` to -1 which is the id of the foundation theme. We should disable the "Theme is enabled by default" checkbox on Foundation in the UI when it's the default theme.

Reply:
I just checked that `Theme.clear_default!` is happening in 2 places:
- when the theme is converted to a component (which is not possible for system themes)
- when another theme is marked as default and ` @theme.id == SiteSetting.default_theme_id`

It feels safe to me.

Reply:
Sorry I wasn't very clear here. My point is about the "Theme is enabled by default" checkbox rather than the `clear_default!` method itself. As an admin, you can turn on this checkbox on the Foundation theme:

![image](https://github.com/user-attachments/assets/4752c797-75d8-4535-9e4e-7453adae0eac)

But once it's on, you can't turn it off from Foundation; you'd have to go to another theme and mark it as the default theme. In a way, trying to remove the Foundation theme as the default doesn't make sense because Foundation === no theme, but my comment is purely a UX comment; there shouldn't be a checkbox that doesn't work.

Reply:
Ahh I see, yeah, definitely we have to improve it. I will ask Ella how to deal with it. Maybe we can drop it for all the themes and relay on option on themes list to mark another theme as default

<img width="1356" alt="Screenshot 2025-06-12 at 11 50 58 am" src="https://github.com/user-attachments/assets/0d68d6ab-7375-49f3-9772-85527ee66dea" />